src/confdb/confdb_setup.c

```diff
@@ -44,6 +44,35 @@
      "\n"
 
 
+errno_t confdb_read_ini(TALLOC_CTX *mem_ctx,
+                     const char *config_file,
+                     const char *config_dir,
+                     bool allow_missing_file,
+                     struct sss_ini **_ini)
+{
+    int ret;
+
+    *_ini = sss_ini_new(mem_ctx);
+    if (*_ini == NULL) {
+        return ENOMEM;
+    }
+
+    ret = sss_ini_read_sssd_conf(*_ini,
+                                 config_file,
+                                 config_dir);
+    if (ret != EOK) {
+        if ((ret == ERR_INI_EMPTY_CONFIG) && allow_missing_file) {
+            return EOK;
+        }
+        talloc_zfree(*_ini);
+        return ret;
+    }
+
+    sss_ini_call_validators(*_ini, SSSDDATADIR"/cfg_rules.ini");
```

Comment:
Shouldn't we do something with the value returned by `sss_ini_call_validators()`?

Reply:
But what?

Original code had:
```
    if (ret != EOK) {
        DEBUG(SSSDBG_CRIT_FAILURE, "Failed to call validators\n");
        /* This is not fatal, continue */
    }
```

Reply:
I removed 'DEBUG' because logger isn't initialized here anyway.

Reply:
You are right. The old code was doing nothing. But then, what's the use of validating the configuration if we'll do nothing about it, not even log a message? Maybe the function should return with the error code and prevent the daemon from starting. I imagine we could also display a message on STDERR, using `ERROR()`.

If we consider that, at least in KCM (and the others should do the same if they aren't already), logger initialization happens very early if the `-d` option is provided, these `DEBUG()` message are still useful when error happens during early initialization. In particular when the daemon fails to start.

Reply:
> what's the use of validating the configuration if we'll do nothing about it, not even log a message?

There are two different things:

(1) invalid configuration;
(2) `sss_ini_call_validators()` failed to run for some reason.

In all cases `sss_ini_call_validators()` logs error itself:
https://github.com/SSSD/sssd/blob/88a386e12a11287771d5429b11b066bf6e75e42f/src/util/sss_ini.c#L782
https://github.com/SSSD/sssd/blob/88a386e12a11287771d5429b11b066bf6e75e42f/src/util/sss_ini.c#L791
https://github.com/SSSD/sssd/blob/88a386e12a11287771d5429b11b066bf6e75e42f/src/util/sss_ini.c#L797

Log I removed didn't add anything.

As to my statement "logger isn't initialized here anyway" - this is incorrect, at least in old version:
```
[sssd] [sss_ini_call_validators] (0x0020): [rule/allowed_sssd_options]: Attribute 'test' is not allowed in section 'sssd'. Check for typos.
```
I'll check if this PR breaks it...

Reply:
It still works:
```
[sssd] [sss_ini_call_validators] (0x0020): [rule/allowed_sssd_options]: Attribute 'test' is not allowed in section 'sssd'. Check for typos.
(2023-10-13 21:25:03): [sssd] [server_setup] (0x3f7c0): Starting with debug level = 0x2f7f0
```

It works because `DEBUG_INIT` is currently done *before* `load_configuration()`.

Reply:
OK

Reply:
In the latest version `confdb_read_ini()` -> `sss_ini_call_validators()` happens **before** `DEBUG_INIT`, so this is probably broken...